packages/env/lib/config.js

```diff
@@ -143,23 +145,8 @@ module.exports = {
 			);
 		}
 
-		if (
-			! Array.isArray( config.plugins ) ||
-			config.plugins.some( ( plugin ) => typeof plugin !== 'string' )
-		) {
-			throw new ValidationError(
-				'Invalid .wp-env.json: "plugins" must be an array of strings.'
-			);
-		}
-
-		if (
-			! Array.isArray( config.themes ) ||
-			config.themes.some( ( theme ) => typeof theme !== 'string' )
-		) {
-			throw new ValidationError(
-				'Invalid .wp-env.json: "themes" must be an array of strings.'
-			);
-		}
+		// Throw an error if any of the sources are invalid.
+		validateSources( config, [ 'plugins', 'themes', 'mu-plugins' ] );
```

Comment:
I get that without this abstraction there's a lot of repetition but, to me, this feels a little _too_ abstract. Maybe a compromise would be to remove the `forEach` and call `validateSources` once per source?

```js
validateSources( config.plugins );
validateSources( config.themes );
validateSources( config[ 'mu-plugins' ] );
```

Reply:
Ok :) This now looks like `validateSources( config.plugins, 'plugins' );` since we need to know "name" of the source type so we can inform the user which property needs to be fixed. An alternative is `validateSources( config, 'plugins' );` (where we would read the value from the config inside the function). What do you think?

Reply:
I prefer how you have it 🙂